python/pyspark/pandas/frame.py

```diff
@@ -3989,9 +3989,14 @@ def insert(
             )
 
         if is_name_like_tuple(column):
-            if len(column) != len(self.columns.levels):
-                # To be consistent with pandas
-                raise ValueError('"column" must have length equal to number of column levels.')
+            if isinstance(self.columns, pd.MultiIndex):
+                if len(column) != len(self.columns.levels):
+                    # To be consistent with pandas
+                    raise ValueError('"column" must have length equal to number of column levels.')
+            else:
+                raise NotImplementedError(
+                    "Tuple-like name is not supported to non-MultiIndex column"
+                )
 
         if column in self.columns:
             raise ValueError("cannot insert %s, already exists" % column)
```

Comment:
Let's explicitly call `str(column)` here. Otherwise, it will fail to format when `column` is a `tuple`. e.g.)

```
    raise ValueError("cannot insert %s, already exists" % column)
TypeError: not all arguments converted during string formatting
```